File: 203-remove-linked-list-elements/remove-linked-list-elements.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

struct ListNode* createNode(int val) {
    struct ListNode* res = malloc(sizeof(struct ListNode));
    res->val = val;
    res->next = NULL;
    return res;
}
struct ListNode* removeElements(struct ListNode* head, int val) {
    struct ListNode* newHead = NULL;
    struct ListNode* curr = newHead;
    struct ListNode* temp = NULL;
    for (struct ListNode* p = head; p; p = p->next) {
        if (p->val == val)
            continue;
        temp = createNode(p->val);
        if (!newHead)
            newHead = temp;
        else
            curr->next = temp;
        curr = temp;
    }
    return newHead;
}

File: 203-remove-linked-list-elements/remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        newHead = None
        curr = newHead
        p = head
        while p:
            if p.val == val:
                p = p.next
                continue
            temp = ListNode(p.val)
            if newHead == None:
                newHead = temp
            else:
                curr.next = temp
            curr = temp
            p = p.next
        return newHead
